src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights.AspNetCore/TargetingTelemetryInitializer.cs

```diff
@@ -42,20 +40,13 @@ protected override void OnInitializeTelemetry(HttpContext httpContext, RequestTe
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
-            // Extract the targeting id from the http context
-            string targetingId = null;
-
-            if (httpContext.Items.TryGetValue(TargetingIdKey, out object value))
-            {
-                targetingId = value?.ToString();
-            }
-
-            if (!string.IsNullOrEmpty(targetingId))
+            // Extract the targeting id from the request telemetry
+            if (requestTelemetry.Properties.ContainsKey(Constants.TargetingIdKey))
```

Comment:
Should we do null check for `requestTelemetry` too?

Reply:
Adjusted the `httpContext` null check to be a check on `requestTelemetry` instead.